umbra-js/src/utils/utils.ts

```diff
@@ -241,6 +243,15 @@ export async function lookupRecipient(
   return { spendingPublicKey: publicKey, viewingPublicKey: publicKey };
 }
 
+export async function getBlockNumberUserRegistered(address: string, startblock = 0, Signer: JsonRpcSigner) {
+  address = getAddress(address); // address input validation
+  const { chainId } = await Signer.provider.getNetwork();
+  const txHistoryProvider = new TxHistoryProvider(chainId);
+  const history = await txHistoryProvider.getHistory(address, startblock);
+  const registryBlock = history.find((tx) => tx.to === STEALTH_KEY_REGISTRY_ADDRESS);
+  return registryBlock?.blockNumber;
+}
```

Comment:
An alternative approach here is query the StealthKeyRegistry for the `StealthKeyChanged`, and filter on the registrant, which is an indexed param. So your log query will return all `StealthKeyChanged` events for that user, and you'd return the oldest block number in that list.

You'd do this with `provider.getLogs(filter)` where the filter is something like: 
```typescript
const filter = stealthKeyRegistry.filters.StealthKeyChanged(user, null, null, null, null)`
```

My gut is to prefer the event approach to minimize our reliance on Etherscan's API, and I suspect it will be fast too because it's going to return a lot less data especially for large accounts.

I also suspect that for users with large numbers of txs, `.getHistory` won't return all of them and there may be some pagination, so this approach might fail sometimes

Reply:
Made the changes here: `provider.getLogs(filter)` didn't seem to return any logs so used `registry._registry.queryFilter(filter)`. Seems to work just as well?

Reply:
This looks great, thanks! I see you use `const registryBlock = stealthKeyLog[0].blockNumber`—I changed the address param to `null` to verify that logs are returned by oldest first, meaning we're correct in using index 0 here, but would great for you to verify too (if you didn't already)

Similarly, a nit: `stealthKeyLogs` (plural) is a better var name since this is an array

Reply:
Thank you! Yes, can confirm the oldest blocks are returned first in the array. Updated var name in https://github.com/ScopeLift/umbra-protocol/pull/549/commits/c7ec660a2d2b4fd0efe601b7374a06597cc7255b